robot/rover/config/.bash_aliases

```diff
@@ -16,9 +16,12 @@ alias arm="cd $ROVER/ArmDriverUnit"
 alias wheels="cd $ROVER/MobilePlatform"
 alias rostings="cd $ROSPACKAGES"
 alias mcunode="cd $ROSPACKAGES/src/mcu_control/scripts"
+alias upload="cmake --build $ROVER/build --target "
```

Comment:
what exactly makes the upload trigger from this command versus doing just `cmake ..`? Is it the --target option? And why is that?

Reply:
Targets are just a fancy name for binaries, libraries, or encapsulations for a set of custom bash scripts or configurations. When you do `cmake ..` the targets aren't actually built, they just configured to be built by either `make` (which by default does NOT build custom targets (although there are tricks around this), only binaries and libraries are built), but to upload to a teensy board, there's no default way of doing it like compiling a C++ project or something, you need external scripts hence custom commands hence the use of custom targets. For this issue, it's also much more convenient to build targets from terminal whenever you want, hence this command, where `--target NAME` builds the custom `NAME` target that was configured by `cmake..` , and `--build` just specifies that something is going to be built that was configured in the `$ROVER/build` directory.